### ISSUE TITLE
implement delete resource with links with acl:write (atomic delete)

### ISSUE THREAD

**Issue by bourgeoa** [jeff-zucker/solid-rest#26]
@jeff-zucker 
Since NSS 5.4.0 solid as implemented atomic delete with `acl:Write`
Actually rest.js only delete the item resource and not all the auxiliary resources (ending with .acl, .meta and .meta.acl, ...)

To delete auxiliary resource you need to delete each resource independently with a need for `acl:Control` if you want to delete an ACL resource
For example this is done in solid-file-client that has implemented `_deleteItemWithLinks()`

```
  async _deleteItemWithLinks (itemUrl) {
    const links = await this.getItemLinks(itemUrl, { links: LINKS.INCLUDE })
    if (links.meta) {
      await this._deleteItemWithLinks(links.meta)
    }
    if (links.acl) {
      await this.delete(links.acl)
    }

    // Note: Deleting item after deleting links to make it work for folders
    //       Change this if a new spec allows to delete them together (to avoid deleting the permissions before the folder)
    return this.delete(itemUrl)
  }

```
The function should be replaced by 
```
  async _deleteItemWithLinks (itemUrl) {
     return this.delete(itemUrl)
  }

```
A bunch of tests are failing because they use solid-rest and solid-rest only mock a simple delete and not an atomic delete
A solution is to implement a deleteItemWithLinks() in solid-rest

**Comment by jeff-zucker** [comment]
Thanks much for the code and tests.  I will npm it shortly.

**Comment by bourgeoa** [comment]
I am unable to test browserFS so please review the code (it replicates the code from `file`)

**Comment by jeff-zucker** [comment]
Understood.  My current inclination is to suspend development of browserFS and concentrate on keeping file and localStorage on the command line up to date but not the browser.  I would mark it as experimental in the docs and return to it eventually when the other things I have responsibility for are done.  Thoughts?

**Comment by jeff-zucker** [comment]
AFAIK, this was solved in PR #27 ,
